### PR TITLE
feat(status): add new status tag component

### PR DESCRIPTION
**Pull request by laraujo7**
2023-01-05-162127_641x1078_scrot

### CODE REVIEW

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]
Please add usage samples in documentation and add a screenshot in PR description

File: packages/ui-components/src/components/status/status.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+import { EStatusType, IStatus } from './status.types';
+import { STATUS_ICON_NAME, STATUS_STATE, STATUS_TEXT } from './status.utils';
+
+@Component({
+	tag: 'kv-status',
+	styleUrl: 'status.scss',
+	shadow: true
+})
+export class KvStatus implements IStatus {
```
Comment (pedrorebelocosta) [inline packages/ui-components/src/components/status/status.tsx:10]: Naming can be a bit confusing here, should we change this to `KvTagStatus` or `KvStatusTag`? 

I believe this naming schema can also be confusing in the props (i.e `type`, where we're actually defining the state of this component) but this is debatable.

**Review by pedrorebelocosta — CHANGES_REQUESTED** [review]
other than the suggestions left by @ricardogoncalves89 please check the comment I've left regarding the naming of this component, good job!

File: packages/ui-components/src/assets/styles/kv-theme.default.scss [hunk 2]
```diff
@@ -95,8 +95,7 @@ $colors: (
 		light: get-color-light($error-color)
 	),
 	'background': (
-		/* Steps move towards their contrasting color. */
-		'step-0': $neutral-color-9,
+		/* Steps move towards their contrasting color. */ 'step-0': $neutral-color-9,
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/assets/styles/kv-theme.default.scss:98]: fix this

File: packages/ui-components/src/assets/styles/kv-theme.default.scss [hunk 3]
```diff
@@ -108,8 +107,7 @@ $colors: (
 		'step-8': $neutral-color-0
 	),
 	'text': (
-		/* Steps move towards their contrasting color. */
-		'step-0': $neutral-color-0,
+		/* Steps move towards their contrasting color. */ 'step-0': $neutral-color-0,
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/assets/styles/kv-theme.default.scss:110]: fix this

**Review by pedrorebelocosta — APPROVED** [review]

File: packages/ui-components/src/components/tag-status/tag-status.types.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+export enum EState {
+	Success = 'success',
+	Error = 'error',
+	Unknown = 'unknown',
+	Warning = 'warning'
+}
+
+export interface IStatusConfig {
+	text: string;
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/tag-status/tag-status.types.ts:26]: Sorry but I think `label` is a better naming here

File: packages/ui-components/src/components/tag-status/tag-status.types.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { EIconName } from '../icon/icon.types';
+
+export interface IStatus {
+	/** (optional) Defines the status tag type.*/
+	type?: EStatusType;
+}
+
+export enum EStatusType {
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/tag-status/tag-status.types.ts:8]: rename to `ETagStatusType`

File: packages/ui-components/src/components/tag-status/tag-status.types.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { EIconName } from '../icon/icon.types';
+
+export interface IStatus {
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/tag-status/tag-status.types.ts:3]: rename to `ITagStatus`

File: packages/ui-components/src/components/tag-status/tag-status.types.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+export enum EState {
+	Success = 'success',
+	Error = 'error',
+	Unknown = 'unknown',
+	Warning = 'warning'
+}
+
+export interface IStatusConfig {
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/tag-status/tag-status.types.ts:25]: rename to `ITagStatusConfig`

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/assets/styles/kv-theme.default.scss (hunk 3 above)
Comment (laraujo7) [reply packages/ui-components/src/assets/styles/kv-theme.default.scss:110]: Sorry about that, my autosave is formating that :pensive:

File: packages/ui-components/src/components/tag-status/tag-status.scss [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+		}
+	}
+
+	&-error {
+		kv-icon {
+			--icon-color: #{kv-color('error')};
+		}
+	}
+
+	&-unknown {
```
Comment (ricardogoncalves89) [inline packages/ui-components/src/components/tag-status/tag-status.scss:26]: I think here should be `-none` but please confirm

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]

File: packages/ui-components/src/components/tag-status/tag-status.scss (hunk 8 above)
Comment (laraujo7) [reply packages/ui-components/src/components/tag-status/tag-status.scss]: Yeap :heavy_check_mark:

**Pull request merged**